packages/frontend/src/pages/SavedExplorer.tsx

```diff
@@ -33,11 +36,15 @@ const SavedExplorer = () => {
     });
 
     useEffect(() => {
+        // If the saved explore is part of a dashboard, set the dashboard chart info
+        // so we can show the banner + the user can navigate back to the dashboard easily
         if (data && data.dashboardUuid && data.dashboardName) {
-            sessionStorage.setItem('fromDashboard', data.dashboardName);
-            sessionStorage.setItem('dashboardUuid', data.dashboardUuid);
+            setDashboardChartInfo({
+                name: data.dashboardName,
+                dashboardUuid: data.dashboardUuid,
+            });
         }
-    }, [data]);
+    }, [data, setDashboardChartInfo]);
```

Comment:
Even though it should have been obvious, I had a bit of a hard time understanding this chunk.

This is for already saved charts - not for charts you are creating or updating from the dashboard - right?

Reply:
Exactly, for already-saved-charts-within-a-dashboard!